web/src/main/java/com/bazaarvoice/emodb/web/compactioncontrol/CompactionControlMonitor.java

```diff
@@ -44,19 +49,32 @@ protected void shutDown() {
     @Override
     protected void runOneIteration() {
         try {
+            updateMetrics(_clock.millis());
             deleteExpiredStashTimes(_clock.millis());
         } catch (Throwable t) {
             _log.error("Unexpected exception.", t);
         }
-        ;
+    }
+
+    private void updateMetrics(long currentTimeInMillis) {
+        Map<String, StashRunTimeInfo> stashTimeInfoMap = _compactionControlSource.getAllStashTimes();
+        if (stashTimeInfoMap.size() > 0) {
+            DateTime oldestCompactionControlTime = new DateTime(stashTimeInfoMap.entrySet().stream()
+                    .min((entry1, entry2) -> entry1.getValue().getTimestamp() > entry2.getValue().getTimestamp() ? 1 : -1)
+                    .get().getValue().getTimestamp());
```

Comment:
Nothing wrong with the above.  Just pointing out that it could be made more readable since `entry.key()` is never used, it's using a natural ordering and it's instantiating a `DateTime` only to call `getMillis()` on it:

```
            long oldestCompactionControlTime = stashTimeInfoMap.values().stream()
                    .map(StashRunTimeInfo::getTimestamp)
                    .min(Ordering.natural())
                    .orElse(currentTimeInMillis);
```